plugins/flutter_aepedgeidentity/android/src/main/java/com/adobe/marketing/mobile/flutter/flutter_aepedgeidentity/FlutterAEPEdgeIdentityPlugin.java

```diff
@@ -0,0 +1,179 @@
+    Identity.getExperienceCloudId(new AdobeCallbackWithError<String>() {
+        @Override
+        public void call(final String experienceCloudId) {
+            AndroidUtil.runOnUIThread(new Runnable() {
+                @Override
+                public void run() {
+                    result.success(experienceCloudId);
+                }
+            });
+        }
+
+        @Override
+        public void fail(final AdobeError adobeError) {
+          final AdobeError error = adobeError != null ? adobeError : AdobeError.UNEXPECTED_ERROR;
+          AndroidUtil.runOnUIThread(new Runnable() {
+            @Override
+            public void run() {
+              result.error(Integer.toString(error.getErrorCode()),"getExperienceCloudId - Failed to retrieve Experience Cloud Id",error.getErrorName());
+            }
+          });
+        } 
+    });
+  }
+
+  private void handlerGetUrlVariables(final MethodChannel.Result result) {
+      Identity.getUrlVariables(new AdobeCallbackWithError<String>() {
+          @Override
+          public void call(final String urlVariables) {
+              AndroidUtil.runOnUIThread(new Runnable() {
+                  @Override
+                  public void run() {
+                      result.success(urlVariables);
+                  }
+              });
+          }
+
+          @Override
+          public void fail(final AdobeError adobeError) {
+            final AdobeError error = adobeError != null ? adobeError : AdobeError.UNEXPECTED_ERROR;
+            AndroidUtil.runOnUIThread(new Runnable() {
+              @Override
+              public void run() {
+                result.error(Integer.toString(error.getErrorCode()),"getExperienceCloudId - Failed to retrieve Experience Cloud Id",error.getErrorName());
+              }
+            });
+          }
+      });
+  }
+
+  private void handlerGetIdentities(final MethodChannel.Result result) {
+    Identity.getIdentities(new AdobeCallbackWithError<IdentityMap>() {
+        @Override
+        public void call(IdentityMap map) {
+            AndroidUtil.runOnUIThread(new Runnable() {
+                @Override
+                public void run() {
+                    Map identityMap = FlutterAEPEdgeIdentityDataBridge.mapFromIdentityMap(map);
+                    result.success(identityMap);
+                }
+            });
+        }
+
+        @Override
+        public void fail(final AdobeError adobeError) {
+          final AdobeError error = adobeError != null ? adobeError : AdobeError.UNEXPECTED_ERROR;
+          AndroidUtil.runOnUIThread(new Runnable() {
+            @Override
+            public void run() {
+              result.error(Integer.toString(error.getErrorCode()),"getIdentity - Failed to retrieve identities",error.getErrorName());
+            }
+          });
+        }
+    });
+  }
+
+ @SuppressLint("LongLogTag")
+  private void handlerUpdateIdentities(final Object arguments) {
```

Comment:
Updating Identities failed, arguments are invalid", what are arguments? it's IdentityItems passed by the customer right?

Reply:
It is identityMap. We can say something like `... passed IdentityMap is invalid`